src/app/trajectory/map/map.page.ts

```diff
@@ -9,63 +20,74 @@ import { InferenceService } from '../inferences/inference.service'
   templateUrl: './map.page.html',
   styleUrls: ['./map.page.scss'],
 })
-export class MapPage implements OnInit {
-  mapOptions: MapOptions
-  map: Map
-  inferences: Inference[]
-  trajectoryId: string
+export class MapPage implements OnInit, OnDestroy {
+  mapOptions: MapOptions = {
+    center: [51.9694, 7.5954],
+    zoom: 14,
+    layers: [
+      tileLayer('https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png', {
+        maxZoom: 19,
+        attribution:
+          '© <a href="http://www.openstreetmap.org/copyright">OpenStreetMap</a>',
+      }),
+    ],
+  }
+  mapBounds: LatLngBounds
+  polyline: Polyline
+  inferenceMarkers = new LayerGroup()
+
+  // should only be used for invalidateSize(), content changes via directive bindings!
+  private map: Map | undefined
+  private trajSub: Subscription
 
   constructor(
-    private service: InferenceService,
-    private route: ActivatedRoute,
-    private trajectoryService: TrajectoryService
+    private inferences: InferenceService,
+    private trajectories: TrajectoryService,
+    private route: ActivatedRoute
   ) {}
 
   ngOnInit() {
-    this.initMapOptions()
-    this.trajectoryId = this.route.snapshot.paramMap.get('trajectoryId')
-    this.inferences = this.service.getInferences(this.trajectoryId)
+    const trajectoryId = this.route.snapshot.paramMap.get('trajectoryId')
+    const trajectoryType = this.route.snapshot.paramMap.get(
+      'trajectoryType'
+    ) as TrajectoryType
+
+    this.trajSub = this.trajectories
+      .getOne(trajectoryType, trajectoryId)
+      .subscribe((t) => {
+        this.polyline = new Polyline(t.coordinates)
+        this.mapBounds = this.polyline.getBounds()
+      })
+
+    this.addInferenceMarkers(this.inferences.getInferences(trajectoryId))
+  }
+
+  ngOnDestroy() {
+    this.trajSub.unsubscribe()
   }
 
   ionViewDidEnter() {
-    this.map.invalidateSize()
-    this.addInferenceMarkers()
-    this.addTrajectory()
+    // required after visibility of map changed.
+    this.map?.invalidateSize()
+
+    // TODO: rework this with optional inference type parameter,
+    //   which we subscribe to and use to set zoom & open popup
+    if (history.state.center)
+      this.mapBounds = latLng(history.state.center).toBounds(100)
```

Comment:
This condition is never satisfied? Or am I missing something?

Reply:
It is when called from https://github.com/sitcomlab/simport-learning-app/blob/89841b5/src/app/trajectory/inferences/inferences.page.ts#L30-L33

Reply:
Ah, that slipped my attention!